packages/block-library/src/search/edit.js

```diff
@@ -76,6 +80,32 @@ export default function SearchEdit( {
 	const borderColor = style?.border?.color;
 	const borderProps = useBorderProps( attributes );
 
+	const { __unstableMarkNextChangeAsNotPersistent } = useDispatch(
+		blockEditorStore
+	);
+
+	// Check if Navigation contains the block.
+	const isNavigationChild = 'showSubmenuIcon' in context;
+	const hasChangedDefaults = useRef( false );
+
+	useEffect( () => {
+		if ( hasChangedDefaults.current ) {
```

Comment:
>so if you basically reset the navigation block to specifically look like the default state, it will reset on reload?

Yes, but most importantly it will update existing `Search` blocks.

I made an alternative here: https://github.com/WordPress/gutenberg/pull/36125